packages/sample-app/src/components/common/LabelWithTooltipIcon.tsx

```diff
@@ -0,0 +1,29 @@
+import { Flex, FlexItem, Tooltip } from '@patternfly/react-core';
+import { InfoCircleIcon } from '@patternfly/react-icons';
+// eslint-disable-next-line camelcase
+import { global_info_color_100 } from '@patternfly/react-tokens';
+import * as React from 'react';
+
+type LabelWithTooltipIconProps = {
+  label: React.ReactNode;
+  tooltipContent?: React.ReactNode;
+};
+
+const LabelWithTooltipIcon: React.FC<LabelWithTooltipIconProps> = ({ label, tooltipContent }) => {
+  if (!tooltipContent) {
+    return label as React.ReactElement;
```

Comment:
This cast was added to work around a TypeScript typing issue.

One solution would be a fragment `<>{label}</>` but this triggers an ESLint warning about avoiding single-element fragments.

On the other hand, typing `label` as `ReactElement` would restrict users from passing e.g. `undefined` or `boolean` or React fragment etc.

This is basically a `ReactNode` input :arrow_right: `React.FC` output passthrough issue.

@florkbr @Hyperkid123 Is there some obvious simple solution to this that I'm missing?

Reply:
Yeah, this is fine. I've encountered this many times. I don't think there is a nice solution to this. Technically `undefined` or `boolean` is not a valid ReactNode. Bot If it was not supposed to be permitted then react should throw a runtime error. I think the type cast is OK.

Reply:
@Hyperkid123 Thanks :+1: I've found some discussions around this issue, for example:

https://github.com/DefinitelyTyped/DefinitelyTyped/issues/18051#issuecomment-645447374

Some devs even advocate not using `React.FC` type at all:

https://kentcdodds.com/blog/how-to-write-a-react-component-in-typescript

The technically correct solution (type wise) would be:

```tsx
// rule suppression below can be avoided with "allowExpressions: true" config
// eslint-disable-next-line react/jsx-no-useless-fragment
return React.isValidElement(label) ? label : <>{label}</>;
```

but that's just too complex for common use, so I'll keep the type cast.

Reply:
Hopefully React 18 has addressed this already :smiley: